### PR TITLE
add unittests and support python3 along with python2

### PR DESCRIPTION
**Pull request by ornoone**
support python 2.7, 3.5, 3.6, 3.7 + django 1.11, 2.0, 2.1, 2.2 + nameko 2.11, 2.12
more unittesting, and a integration test with a running service + rabbitmq
add tox to test on all versions
add coverage
integration with tarvis-ci parallel run for all python versions
add flake8 and isort

### CODE REVIEW

File: django_nameko/rpc.py [hunk 1]
```diff
@@ -93,8 +97,9 @@ def __exit__(self, exc_type, exc_value, traceback, **kwargs):
                             for key in self._rpc._worker_ctx.data.keys():
```
Comment (ornoone) [inline django_nameko/rpc.py:97]: it's not clear what this code do, and how to cover it with unittests.

File: django_nameko/rpc.py [hunk 2]
```diff
@@ -218,43 +225,41 @@ def get_pool(pool_name=None):
...
-                    # assign nameko_global_pools to corresponding name
-                    nameko_global_pools[name] = _pool
-            else:
-                # single nameko_global_pools with old style configuration
-                if not hasattr(settings, 'NAMEKO_CONFIG') or not settings.NAMEKO_CONFIG:
-                    raise ImproperlyConfigured(
-                        'NAMEKO_CONFIG must be specified and should include at least "AMQP_URL" key.')
-                nameko_global_pools = ClusterRpcProxyPool(settings.NAMEKO_CONFIG)
-                nameko_global_pools.start()  # start immediately
+        with create_pool_lock:
```
Comment (ornoone) [inline django_nameko/rpc.py:228]: suround with the contextmanager use of the lock. it prevent a bug: when the creation of the cluster raise a Exception, the lock.release was never called.

File: django_nameko/rpc.py [hunk 3]
```diff
@@ -218,43 +225,41 @@ def get_pool(pool_name=None):
...
+            if not nameko_global_pools:  # double check inside lock is importance
+                if NAMEKO_MULTI_POOL:
+                    nameko_global_pools = dict()
+                    if 'default' not in NAMEKO_CONFIG or 'AMQP_URL' not in NAMEKO_CONFIG['default']:
```
Comment (ornoone) [inline django_nameko/rpc.py:232]: fix the `and` into a `or`

File: django_nameko/rpc.py [hunk 4]
```diff
@@ -264,9 +269,10 @@ def get_pool(pool_name=None):
             _pool = nameko_global_pools[pool_name]
     else:
         if isinstance(nameko_global_pools, dict):
-            if len(nameko_global_pools) == 0:
+            if len(nameko_global_pools) == 0:  # pragma: nocover
+                # this code is unreachable, it's not passilbe to have a dict without a key in it.
                 raise ImproperlyConfigured('NAMEKO_CONFIG must include at least 1 "default" config')
-            _pool = nameko_global_pools.get('default', nameko_global_pools.values()[0])
+            _pool = nameko_global_pools.get('default', next(iter(nameko_global_pools.values())))
```
Comment (ornoone) [inline django_nameko/rpc.py:275]: fix for python3

File: django_nameko/rpc.py (hunk 2 above)
Comment (0xGosu) [reply django_nameko/rpc.py:228]: agree, i found this bug also. But haven't fix this on main repo

File: django_nameko/rpc.py (hunk 3 above)
Comment (0xGosu) [reply django_nameko/rpc.py:232]: agree, correct fix

File: django_nameko/rpc.py (hunk 1 above)
Comment (0xGosu) [reply django_nameko/rpc.py:97]: this part of the code is for checking context data of the worker after RPC is complete and make sure it is rever back to original pool context data

**Review by 0xGosu — COMMENTED** [review]
I will check your PR and come back with more comment

File: django_nameko/rpc.py (hunk 1 above)
Comment (ornoone) [reply django_nameko/rpc.py:97]: can you give me a snipet of code where the worker context is altered ? I did not succeed in doing it in the tests.

**Comment by ornoone** [comment]
as a side note, I added coverage support for the project. you can enable it before the merge, this way it will trigger the first coverage during the merge. 

you just have to go to https://coveralls.io/repos/new and add django-nameko and then merge this pr.

**Pull request merged**